openedx/features/enterprise_support/api.py

```diff
@@ -358,28 +361,63 @@ def get_enterprise_customer_cache_key(uuid, username=settings.ENTERPRISE_SERVICE
     )
 
 
-@enterprise_is_enabled()
 def cache_enterprise(enterprise_customer):
-    """Cache this customer's data."""
+    """Add this customer's data to the Django cache."""
     cache_key = get_enterprise_customer_cache_key(enterprise_customer['uuid'])
     cache.set(cache_key, enterprise_customer, settings.ENTERPRISE_API_CACHE_TIMEOUT)
 
 
-@enterprise_is_enabled()
-def enterprise_customer_from_cache(request=None, uuid=None):
-    """Check all available caches for Enterprise Customer data."""
-    enterprise_customer = None
+def enterprise_customer_from_cache(uuid):
+    """
+    Retrieve enterprise customer data associated with the given ``uuid`` from the Django cache,
+    returning a ``__CACHE_MISS__`` if absent.
+    """
+    cache_key = get_enterprise_customer_cache_key(uuid)
+    return cache.get(cache_key, _CACHE_MISS)
 
-    # Check if it's cached in the general cache storage.
-    if uuid:
-        cache_key = get_enterprise_customer_cache_key(uuid)
-        enterprise_customer = cache.get(cache_key)
 
-    # Check if it's cached in the session.
-    if not enterprise_customer and request:
-        enterprise_customer = request.session.get('enterprise_customer')
+def add_enterprise_customer_to_session(request, enterprise_customer):
+    """ Add the given enterprise_customer data to the request's session. """
+    request.session[ENTERPRISE_CUSTOMER_KEY_NAME] = enterprise_customer
 
-    return enterprise_customer
+
+def enterprise_customer_from_session(request):
+    """
+    Retrieve enterprise_customer data from the request's session,
+    returning a ``__CACHE_MISS__`` if absent.
+    """
+    return request.session.get(ENTERPRISE_CUSTOMER_KEY_NAME, _CACHE_MISS)
+
+
+def enterprise_customer_uuid_from_session(request):
+    """
+    Retrieve an enterprise customer UUID from the request's session,
+    returning a ``__CACHE_MISS__`` if absent.  Note that this may
+    return ``None``, which indicates that we've previously looked
+    for an associated customer for this request's user, and
+    none was present.
+    """
+    customer_data = enterprise_customer_from_session(request)
+    if customer_data is not _CACHE_MISS:
+        customer_data = customer_data or {}
+        return customer_data.get('uuid')
+    return _CACHE_MISS
+
+
+def enterprise_customer_uuid_from_query_param(request):
+    """
+    Returns an enterprise customer UUID from the given request's GET data,
+    or ``__CACHE_MISS__`` if not present.
+    """
+    return request.GET.get(ENTERPRISE_CUSTOMER_KEY_NAME, _CACHE_MISS)
```

Comment:
`cache_miss` for checking query parameters or cookies feels a little weird to me. I get why you're doing it though, so I think you can ignore me

Reply:
Yup :)